controllers/gcpcluster_controller.go

```diff
@@ -192,8 +192,20 @@ func (r *GCPClusterReconciler) reconcile(clusterScope *scope.ClusterScope) (ctrl
 	}
 	gcpCluster.Status.FailureDomains = make(clusterv1.FailureDomains, len(zones))
 	for _, zone := range zones {
-		gcpCluster.Status.FailureDomains[zone] = clusterv1.FailureDomainSpec{
-			ControlPlane: true,
+		if len(gcpCluster.Spec.FailureDomains) > 0 {
+			found := false
+			for _, fd := range gcpCluster.Spec.FailureDomains {
```

Comment:
I'm not quite sure this will have the desired effect as written. This would still allow for all failure domains in a given zone to get added, just the ones specifed in the spec wouldn't be eligible for controlplane scheduling. 

Wouldn't we rather _only_ add the failuredomains supplied in the spec to the list in the status field?

Reply:
I'm thinking something more like:
```bash
gcpCluster.Status.FailureDomains = make(clusterv1.FailureDomains, len(zones))

// Iterate through all zones
for _, zone := range zones {

	// If we have failuredomains in spec, see if this zone is in that list
	// Add to the status _only_ if it's mentioned in the gcpCluster spec
	if len(gcpCluster.Spec.FailureDomains) > 0 {
		for _, fd := range gcpCluster.Spec.FailureDomains {
			if fd == zone {
				gcpCluster.Status.FailureDomains[zone] = clusterv1.FailureDomainSpec{
					ControlPlane: true,
				}
			}
		}
	// Default to adding all zones in failuredomain status
	} else {
		gcpCluster.Status.FailureDomains[zone] = clusterv1.FailureDomainSpec{
			ControlPlane: true,
		}
	}
}
```

Reply:
++ack. Will update the branch.